hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/source/StreamReadMonitoringFunction.java

```diff
@@ -213,28 +244,47 @@ public void monitorDirAndForwardSplits(SourceContext<MergeOnReadInputSplit> cont
       // table does not exist
       return;
     }
-    IncrementalInputSplits.Result result =
-        incrementalInputSplits.inputSplits(metaClient, this.issuedOffset, this.cdcEnabled);
+
+    IncrementalInputSplits.Result result = remainingSplits.isEmpty()
+        ? incrementalInputSplits.inputSplits(metaClient, this.issuedOffset, this.cdcEnabled)
```

Comment:
IIUC, it's here where performance bottleneck lies, so the problem still exists w/ this PR if one commit contains excessive amount of input splits?

Reply:
yes, so it's little confusing that the issues listed in the description are actually not solved.

Reply:
@fhan688 would u mind to describe what this PR is going to do more accurately? Thanks